shared/settings.cpp

```diff
@@ -0,0 +1,61 @@
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <QDebug>
+#include <QFile>
+#include "settings.h"
+
+Settings::Settings(): _deviceType(DeviceType::RM1) {
+    readDeviceType();
+}
+
+void Settings::readDeviceType() {
+    QFile file("/sys/devices/soc0/machine");
+    if(!file.open(QIODevice::ReadOnly | QIODevice::Text)){
+        qDebug() << "Couldn't open " << file.fileName();
+        return;
+    }
+    QTextStream in(&file);
+    std::string modelName = in.readLine().toStdString();
+    if (modelName.rfind("reMarkable 2", 0) == 0) {
```

Comment:
[QTextStream.readLine()](https://doc.qt.io/qt-5/qtextstream.html#readLine) returns a QString, so you could instead use the following:
```suggestion
    QTextStream in(&file);
    QString modelName = in.readLine();
    if (modelName.startsWith("reMarkable 2")) {
```